node/src/components/fetcher.rs

```diff
@@ -603,6 +603,42 @@ impl ItemFetcher<BlockHeader> for Fetcher<BlockHeader> {
     }
 }
 
+impl ItemFetcher<BlockHeadersBatch> for Fetcher<BlockHeadersBatch> {
+    const SAFE_TO_RESPOND_TO_ALL: bool = true;
+
+    fn responders(
+        &mut self,
+    ) -> &mut HashMap<BlockHeadersBatchId, HashMap<NodeId, Vec<FetchResponder<BlockHeadersBatch>>>>
+    {
+        &mut self.responders
+    }
+
+    fn metrics(&mut self) -> &Metrics {
+        &self.metrics
+    }
+
+    fn peer_timeout(&self) -> Duration {
+        self.get_from_peer_timeout
+    }
```

Comment:
(I wonder whether this trait is a bit redundant: Would it be simpler to move these three getters into `impl<T> Fetcher<T>`, and `get_from_storage` to `Item`? Anyway, has nothing to do with this PR.)